core/src/main/java/com/microsoft/applicationinsights/internal/agent/CoreAgentNotificationsHandler.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * ApplicationInsights-Java
+ * Copyright (c) Microsoft Corporation
+ * All rights reserved.
+ *
+ * MIT License
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ * software and associated documentation files (the ""Software""), to deal in the Software
+ * without restriction, including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.microsoft.applicationinsights.internal.agent;
+
+import java.net.URL;
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.LinkedList;
+
+import com.microsoft.applicationinsights.agent.internal.coresync.AgentNotificationsHandler;
+import com.microsoft.applicationinsights.agent.internal.coresync.InstrumentedClassType;
+
+import com.microsoft.applicationinsights.TelemetryClient;
+import com.microsoft.applicationinsights.internal.logger.InternalLogger;
+import com.microsoft.applicationinsights.internal.schemav2.DependencyKind;
+import com.microsoft.applicationinsights.telemetry.Duration;
+import com.microsoft.applicationinsights.telemetry.RemoteDependencyTelemetry;
+
+/**
+ * The Core's implementation: the methods are called for instrumented methods.
+ * The implementation can measure time in nano seconds, fetch Sql/Http data and report exceptions
+ *
+ * Created by gupele on 5/7/2015.
+ */
+final class CoreAgentNotificationsHandler implements AgentNotificationsHandler {
+
+    /**
+     * The class holds the data gathered on a method
+     */
+    private static class MethodData {
+        public String name;
+        public String[] arguments;
+        public long interval;
+        public InstrumentedClassType type;
+        public Object result;
+    }
+
+    private static class ThreadData {
+        public final LinkedList<MethodData> methods = new LinkedList<MethodData>();
+
+        public void done() {
```

Comment:
It doesn't make much sense to me calling `threadData.done`.
Why not calling this method `reset` or `clean`?